src/components/ProductCard/ProductCard.component.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect } from 'react';
+
+import { CartIcon } from '@/components/CartIcon/CartIcon.component.tsx';
+import type { Product } from '@/models/product.model.ts';
+
+import styles from './ProductCard.module.css';
+
+interface ProductCardProps {
+    product: Product;
+    onCartChange: () => void;
+    cartIds: string[];
+    setProductIds: (ids: string[]) => void;
+}
+
+export const ProductCard: React.FC<ProductCardProps> = ({ product, onCartChange, cartIds, setProductIds }) => {
+    const isInCart = cartIds.includes(product.id.toString());
```

Comment:
I believe that it is necessary to add the possibility for the user to select several items of the same product.

Could you please find out how to rewrite your logic in this component?